### PR TITLE
fixed the deprecated `as_slice` warning...

### PR DESCRIPTION
**Pull request by ghost**
I've put its original implementation from [`core/option.rs`](http://doc.servo.org/src/core/option.rs.html#692) instead of the dear departed `as_slice`

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7741)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Oh no, not yet!

**Comment by Wafflespeanut** [comment]
Hmm, we could have something else...

**Comment by jdm** [comment]
Why not call slice::ref_slice instead of inlining it?

**Comment by metajack** [comment]
Replacing safe code with unsafe code seems not great. This is making the code worse.

What other workarounds are available for the lack of as_slice?

**Comment by Wafflespeanut** [comment]
Yep, I should've gone for `ref_slice`.

File: components/script/dom/websocket.rs [hunk 1]
```diff
@@ -139,7 +139,15 @@ impl WebSocket {
...
 
         // Step 4.
-        let protocols = protocols.as_slice();
+        let protocols = match protocols {
+            Some(ref string) => {
+                slice::ref_slice(string)
+            },
+            None => {
+                let empty_arr: &[_] = &[];
+                empty_arr
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Can these two lines be combined into a single `&[]` expression?

Comment (Wafflespeanut) [inline components/script/dom/websocket.rs]: Of course...

**Comment by jdm** [comment]
@bors-servo: r+
Lovely!

**Comment by Wafflespeanut** [comment]
Thanks! :)

**Pull request merged**